index.js

```diff
@@ -1,18 +1,56 @@
 import fetch from 'node-fetch';
-// Recommend using node-fetch for those familiar with JS fetch
 
 const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
 
 /**
- * @param name filter for color name
- * @param hex filter for color hex code
- * @param compName filter for complementary color name
- * @param compHex filter for complementary color hex code
- * @returns Promise
+ * Fetches and filters colors from the COLORS endpoint.
+ *
+ * @param {Object} filters - Object containing filter criteria.
+ * @param {string} [filters.name] - Filters colors by name (case insensitive).
+ * @param {string} [filters.hex] - Filters colors by exact hex code (without '#').
+ * @param {string} [filters.compName] - Filters colors whose complementary color's name contains this string (case insensitive).
+ * @param {string} [filters.compHex] - Filters colors whose complementary color has this exact hex code (without '#').
+ * @returns {Promise<Array>} - A promise that resolves to an array of filtered colors.
  */
-const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+const fetchColors = async ({ name, hex, compName, compHex }) => {
+  try {
+    // Fetch colors from the COLORS endpoint
+    const response = await fetch(COLORS);
+    const colors = await response.json();
+
+    // 1. Filter by color name (case insensitive)
+    if (name) {
+      return colors.filter(color => color.name.toLowerCase().includes(name.toLowerCase()));
+    }
+
+    // 2. Filter by color hex code (without '#')
+    if (hex) {
+      return colors.filter(color => color.hex.toLowerCase() === hex.toLowerCase());
+    }
+
+    // 3. Filter by complementary color's name (case insensitive)
```

Comment:
you could move all the checks into a single filter function. This way if down the line you want to combine filters, you can do so in the same function.